### ISSUE TITLE
Confusion about symmetric KL loss

### ISSUE THREAD

**Issue by YAaNnGg** [gudovskiy/autodo#1]
Hi! Thanks for your inspiring work.

I am confused about this description in your paper. It's true that we simultaneously optimize conditional distribution and joint distribution, but I think using asymmetric KL loss is ok because the optimizations are in separate steps. Could you please give a further explanation to help me understand it better?

![image](https://user-images.githubusercontent.com/40057959/113732917-917a5200-972c-11eb-81cc-b007e09436a3.png)

Thanks a lot!

**Comment by gudovskiy** [comment]
@YAaNnGg I tried asymmetric KL initially and found some instabilities for certain datasets. After that I decided to address that problem formulation and ended up with symmetric loss. There is a self.sym condition in LossModel class that enables/disables symmetric loss. It could be on or off only if we optimize soft-labels.
Also, after I implemented symmetric loss I found that in the other papers on noisy labels like [this one](https://cs.nju.edu.cn/wujx/paper/PENCIL_CVPR2019.pdf) authored mentioned that usual asymmetric KL may not work and they interchanged distributions in loss. I think, symmetric should be better since we optimize both data and labels at the same time. Let me know if you have better ideas
